.jenkins/actions/build_bare_metal.sh

```diff
@@ -0,0 +1,186 @@
+set +e
+. ${envloc}/env/machineEnvironment.sh
+set -e
+
+# load module tools
+. ${envloc}/env/moduleTools.sh
+
+# load git tools
+. ${envloc}/env/gitTools.sh
+
+# load scheduler tools
+. ${envloc}/env/schedulerTools.sh
+
+# compile the model
+# note: this relies on the fact that daint_gnu and daint_intel are the first
+#       two options for the configure script
+echo "### compile model"
+
+compiler_number=0
+case $compiler in
+  gnu) compiler_number=1 ;;
+  intel) compiler_number=2 ;;
+esac
+cd ${rootdir}/FV3
+echo "${compiler_number}" | ./configure
+
+module unload cudatoolkit
+module unload Boost
+module unload cray-python
+module switch PrgEnv-gnu PrgEnv-cray
+echo "==== module list ===="
+module list
+echo "====================="
+
+./compile
+
+num_exe=`/bin/ls -1d *.exe | wc -l`
+if [ "$num_exe" -lt 1 ] ; then
+   exitError 700 ${LINENO} "No valid executables have been found after compilation"
+fi
+cd -
+
+# install and run example
+# note: we setup the rundir using fv3config in a separate script in order to keep
+#       the environment of this script clean (no modules loaded etc.)
+echo "### run install and example"
+
+script=/tmp/create_rundir_$$.sh
+for config in c12_6ranks_standard c48_6ranks_standard ; do
+    cat > ${script} <<EOF
+#!/bin/bash
+set -e
+export GOOGLE_APPLICATION_CREDENTIALS=/users/olifu/.gc_cred/jenkins-sa-key.json
+cd ${rootdir}/tests/serialized_test_data_generation/
+source ${FV3CONFIG_VENV}/bin/activate
+module load gcloud
+EXPERIMENT=${config} make setup_rundir
+deactivate
+cd -
+EOF
+    chmod 755 ${script}
+    ${script}
+
+    cd ${rootdir}/tests/serialized_test_data_generation/rundir/${config}/
+
+    cp ${rootdir}/FV3/*.exe ./
+    cp ${rootdir}/FV3/conf/modules.fv3 ./module.env
+
+    sed -i 's|^ *months *= *[0-9][0-9]* *$|months = 0|g' input.nml
+    sed -i 's|^ *days *= *[0-9][0-9]* *$|days = 1|g' input.nml
+    sed -i 's|^ *hours *= *[0-9][0-9]* *$|hours = 0|g' input.nml
+    sed -i 's|^ *minutes *= *[0-9][0-9]* *$|minutes = 0|g' input.nml
+    sed -i 's|^ *seconds *= *[0-9][0-9]* *$|seconds = 0|g' input.nml
+
+    jobfile=job
+    cp ${envloc}/env/submit.${host}.${scheduler}  ./${jobfile}
+    sed -i 's|<OUTFILE>|'"${jobfile}.out"'|g' ${jobfile}
+    sed -i 's|<G2G>|'"source ./module.env; module list"'|g' ${jobfile}
+    sed -i 's|<CMD>|'"srun -n 6 ./fv3_64bit.exe"'|g' ${jobfile}
+    sed -i 's|<NAME>|'"fv3_test"'|g' ${jobfile}
+    sed -i 's|<NTASKS>|12|g' ${jobfile}
+    sed -i 's|<NTASKSPERNODE>|'"12"'|g' ${jobfile}
+    sed -i 's|<CPUSPERTASK>|1|g' ${jobfile}
+
+    set +e
+    launch_job ${jobfile} 3000
+    if [ $? -ne 0 ] ; then
+        exitError 710 ${LINENO} "Problem with SLURM job (`pwd`/${jobfile}) see log (`pwd`/${jobfile}.out)"
+    fi
+    grep '^Termination ' job.out > /dev/null
+    if [ $? -ne 0 ] ; then
+        exitError 715 ${LINENO} "Configuration ${config} did not run through (see `pwd`/${jobfile}.out)"
+    fi
+    set -e
+
+done
```

Comment:
Please add some kind of call to `make validate_data` to check that the output data has not changed. This should not be high-effort to add, if it is we can reconsider.

I would suggest re-naming the file to reflect the fact that this tests only the validation data generation. It does not run the tests in `tests/pytest/test_regression.py` which ensure the Fortran model output hasn't changed. Alternatively, you could add a comment with this information to top of the file.

This testing could be added (maybe in another PR) if you're interested. If you implement the sed/slurm logic as a function `run_model_slurm` in test_regression.py, and add an `image_runner` argument to the if-elif-else in `def test_regression`, you can re-use the run directory creation logic and output md5sum check logic we have implemented in that test. This would test both the serialized data output as well as the Fortran model output.

Reply:
You'd likely also need to add regression files for both gnu and intel daint compilations

Reply:
@mcgibbon 

Agreed on the validation. This is a bit tricky since it adds yet another place where we store a reference and since we are building different variants (compiler) and running a different setup (1 day long simulation) on a different computer we will be getting a range of reference files that need to be maintained. I'll think about it when I refactor.

Not sure I understand the naming comment. The goal of this action is to build executables on Piz Daint bare metal. As an add-on, we actually make sure the executables run a test case successfully. Not sure what you mean by "this tests only the validation data generation".

Maybe we should discuss @mcgibbon so that I better understand what you are suggesting here!